src/reactive-handler.ts

```diff
@@ -1,68 +1,62 @@
 import {
     toString,
-    isUndefined,
-    unwrap,
     ArrayConcat,
     isArray,
-    ObjectDefineProperty,
     getPrototypeOf,
-    isExtensible,
-    getOwnPropertyDescriptor,
     getOwnPropertyNames,
     getOwnPropertySymbols,
-    preventExtensions,
-    hasOwnProperty,
+    unwrap,
 } from './shared';
 
 import {
     ReactiveMembrane,
     ReactiveMembraneShadowTarget,
-    wrapDescriptor,
+    isExtensibleMembraneTrap,
+    getOwnPropertyDescriptorMembraneTrap,
+    definePropertyMembraneTrap,
+    preventExtensionsMembraneTrap,
+    MembraneProxyHandler,
 } from './reactive-membrane';
 
-function wrapValue(membrane: ReactiveMembrane, value: any): any {
-    return membrane.valueIsObservable(value) ? membrane.getProxy(value) : value;
-}
-
-/**
- * Unwrap property descriptors will set value on original descriptor
- * We only need to unwrap if value is specified
- * @param descriptor external descrpitor provided to define new property on original value
- */
-function unwrapDescriptor(descriptor: PropertyDescriptor): PropertyDescriptor {
-    if (hasOwnProperty.call(descriptor, 'value')) {
-        descriptor.value = unwrap(descriptor.value);
-    }
-    return descriptor;
-}
-
-function lockShadowTarget(membrane: ReactiveMembrane, shadowTarget: ReactiveMembraneShadowTarget, originalTarget: any): void {
-    const targetKeys = ArrayConcat.call(getOwnPropertyNames(originalTarget), getOwnPropertySymbols(originalTarget));
-    targetKeys.forEach((key: PropertyKey) => {
-        let descriptor = getOwnPropertyDescriptor(originalTarget, key) as PropertyDescriptor;
-
-        // We do not need to wrap the descriptor if configurable
-        // Because we can deal with wrapping it when user goes through
-        // Get own property descriptor. There is also a chance that this descriptor
-        // could change sometime in the future, so we can defer wrapping
-        // until we need to
-        if (!descriptor.configurable) {
-            descriptor = wrapDescriptor(membrane, descriptor, wrapValue);
-        }
-        ObjectDefineProperty(shadowTarget, key, descriptor);
-    });
-
-    preventExtensions(shadowTarget);
-}
+const getterMap = new WeakMap<() => any, () => any>();
```

Comment:
these maps allow us to preserve the identity of the accessors instead of creating new ones every time you access them.